envoy/pkg/livenessprobe/cmd/root.go

```diff
@@ -0,0 +1,168 @@
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/cybozu-go/log"
+	"github.com/cybozu-go/well"
+	"github.com/spf13/cobra"
+)
+
+var config struct {
+	timeout    time.Duration
+	listenAddr string
+	readyURL   string
+	httpURL    string
+	httpsAddr  string
+}
+
+var rootCmd = &cobra.Command{
+	Use:   "livenessprobe",
+	Short: "liveness probe for Envoy",
+	Long:  `Liveness probe for Envoy.`,
+
+	PersistentPreRun: func(cmd *cobra.Command, args []string) {
+		cmd.SilenceUsage = true
+
+		err := well.LogConfig{}.Apply()
+		if err != nil {
+			log.ErrorExit(err)
+		}
+	},
+
+	RunE: func(cmd *cobra.Command, args []string) error {
+		mux := http.NewServeMux()
+		mux.HandleFunc("/", handler)
+		serv := &well.HTTPServer{
+			Server: &http.Server{
+				Addr:    config.listenAddr,
+				Handler: mux,
+			},
+		}
+		err := serv.ListenAndServe()
+		if err != nil {
+			return err
+		}
+
+		err = well.Wait()
+		if err != nil && !well.IsSignaled(err) {
+			return err
+		}
+		return nil
+	},
+}
+
+// Execute executes the command.
+func Execute() {
+	if err := rootCmd.Execute(); err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+}
+
+func handler(rw http.ResponseWriter, req *http.Request) {
+	env := well.NewEnvironment(req.Context())
+	env.Go(monitorReady)
+	env.Go(monitorHTTP)
+	env.Go(monitorHTTPS)
+
+	env.Stop()
+	err := env.Wait()
+	if err != nil {
+		rw.WriteHeader(http.StatusBadGateway)
+		rw.Write([]byte(err.Error()))
+		return
+	}
+
+	rw.WriteHeader(http.StatusOK)
```

Comment:
No need for this.
`Write` w/o calling `WriteHeader` will return response 200.
In fact, both `WriteHeader` and `Write` are not necessary.